From 08387212a0e9d06314ae185935f2756e6db2c70d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?= <avarab@gmail.com>
Date: Sat, 24 Jul 2010 20:05:33 +0000
Subject: [PATCH v2 5/7] Makefile: Add coverage-report-cover-db-html target
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a target to generate a detailed HTML report for the entire Git
codebase using Devel::Cover's cover(1) tool. Output it in
cover_db_html instead of the default cover_db, so that it isn't mixed
up with our raw report files.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 .gitignore |    1 +
 Makefile   |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/.gitignore b/.gitignore
index bd53c02..baed247 100644
--- a/.gitignore
+++ b/.gitignore
@@ -209,3 +209,4 @@
 *.gcov
 /coverage-untested-functions
 /cover_db
+/cover_db_html
diff --git a/Makefile b/Makefile
index f2c680d..b6975aa 100644
--- a/Makefile
+++ b/Makefile
@@ -2307,3 +2307,5 @@ coverage-untested-functions: coverage-report
 coverage-report-cover-db: coverage-report
 	gcov2perl -db cover_db *.gcov
 
+coverage-report-cover-db-html: coverage-report-cover-db
+	cover -report html -outputdir cover_db_html cover_db
-- 
1.7.0.4

